<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Tools for ChIP-Seq and Beyond - Mar 19 2015</title>

		<meta name="description" content="An introduction to Python for Coding">
		<meta name="author" content="Wayne ecatur">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/black.css" id="theme">

		<!-- Code syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">
				<section>
					<h1>A Data Scientist's Toolbox for ChIP-Seq and Beyond</h1>
					<h3>March 19th, 2015</h3>
					<p>
						<small>Created by <a href="http://proteopedia.org/wiki/index.php/User:Wayne_Decatur">Wayne Decatur</a> for Feng Lab Group meeting</small>
					</p>
				</section>

				<section data-markdown>
					<script type="text/template">
						## Session Infrastructure
						[the main portal is here](http://bit.ly/FengMarLM)

						Easy to arrive at two ways:

						- write down `http://bit.ly/FengMarLM`
						- search for `Wayne Decatur proteopedia`, follow github and look for repo for Feng lab group meeting in March (titled 'March 2015 Feng Lab group meeting')

					</script>
				</section>

				<section data-transition="slide" data-background="#4d7e65" data-background-transition="zoom">
					<img data-src="http://image.slidesharecdn.com/2014-bosc-keynote-140711070550-phpapp01/95/2014-bosckeynote-5-638.jpg" alt="Datapocalypse">
					<br>
					borrowed from <a href="https://www.youtube.com/watch?v=uwsjwMO-TEA">A retrospective from the Year 2039 given by Titus Brown</a>

				</section>



				<section data-markdown>
					<script type="text/template">
						### Today is not just about ChIP-seq

						- working productively
						- automation
						- reusable code
						- modular design
						- develop smart, test often
						- programming concepts and algorithmic thinking
						- troubleshooting
						- collaboration
						- reproducibility --> reproducible science
						- open source and open science

							... Datapocalypse (yours and the field in general)

					</script>
				</section>









				<section>
					<h2>FULL DISCLOSURE</h2>
						<p class="fragment"><a href="http://bioinformatics.msu.edu/ngs-summer-course-2014">background</a></p>
				</section>

				<section data-markdown>
					<script type="text/template">
						### open source software
						### open science
						### open access

						filtered views of [Titus Brown and his colleagues such as Stephen Turner, Jonathan Eisen, etc.]()

						<p class="fragment">Not everyone is; your mileage may vary</p>

					</script>
				</section>


				<section>
					<h2>FULL DISCLOSURE</h2>
						<p class="fragment">Not a bioinformatician</q></p>
						<p class="fragment">Your mileage may vary</p>
				</section>

				<section data-markdown>
					<script type="text/template">
						### open source software
						### open science
						### open access

						filtered views of [Titus Brown and his colleagues such as Stephen Turner, Jonathan Eisen, etc.]()

						Not everyone is; your mileage may vary

					</script>
				</section>





				<section data-markdown>
					<script type="text/template">
						##Data management

						[NSF mandates it for proposals](http://www.nsf.gov/eng/general/dmp.jsp)

						[NIH has a related concept of Data Sharing Policy](http://grants.nih.gov/grants/policy/data_sharing/data_sharing_guidance.Htm)

						[Titus Browns' Data Management Plan - a satire](http://ivory.idyll.org/blog/data-management.html)

					</script>
				</section>


				<section data-markdown>
					<script type="text/template">
						##Today

						- Focus is on preparation for running ChIP-seq pipeline next Session
						- I'll sprinkle in introducing more data tols

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##A Molecular Biologist's Computational Toolbox

						a good selection

						- shell (UNIX command line)
						- regular expressions
						- IPython Notebooks (Python)
						- Git/Github
						- R

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Today

						aside from ChIP-seq

						- regular expressions
						- IPython Notebooks
						- Git/Github
						- R

					</script>
				</section>


				<section data-markdown>
					<script type="text/template">
						##ChIP-seq

						for convenience relied on core of [Istvan Albert's 2012 ChIP-Seq lecture](http://ged.msu.edu/angus/tutorials-2012/day5.html)

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						the overall approach

						map reads --> call peaks

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						example and opportunity to add a tool to your toolbelt

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						###Regular Expressions

						[Link to hands-on session](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/in%20the%20wild%201/)

					</script>
				</section>


				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						NUCwave
						ChIP-Seq single reads --> map reaps to genome with Bowtie ---> NUCwave to generate nucleosome occupation maps (peaks)

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						overall approach similar

						map reads --> call peaks

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						most common, probably...

						map reads with bowtie or bwa --> use MACS call peaks

						peaks are the enriched regions

						MACS is python based command line tool

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						full command line workflow example:

						[Hands-on module 4: "ChIP-seq data analysis using command line tools"
This practical tutorial covers alignment, peak calling, annotation and motif discovery of ChIP-seq data.](http://web.csc.fi/english/csc/courses/archive/chipseq2014)

>The analysis tools used in the exercises include several command line tools such as BWA, MACS, MEME and Homer



					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						Using R/Bioconductor popular too

						Another tool...

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##R, Bioconductor project, and RStudio

						[Link](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/in%20the%20wild%204/)

						R is very much about the concept of [tidy data](http://vita.had.co.nz/papers/tidy-data.pdf)

						RStudio can easily be run on Amazon Web Services if computing power is an issue.

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						Using R/Bioconductor popular too

						- do peak calling, annotation, motif discovery and differential binding analysis of ChIP-seq data with  R/Bioconductor packages such as chipseq, Biostrings, BSgenome, DESeq and Diffbind.


					</script>
				</section>


				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						[Analysis of ChIP-seq data in Galaxy](http://jura.wi.mit.edu/bio/education/hot_topics/galaxy/GalaxyNov2012_ChIP-seq_toPost.pdf)

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						Chipster is a user friendly program

						- [covers quality control, alignment, peak calling, annotation, motif discovery](http://web.csc.fi/english/csc/courses/archive/chipseq2014)

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Flavors of basic ChIP-seq analysis

						If your data is Illumina-derived, probably BaseSpace will eventually be an option.

						As far as I know it isn't yet but they keep trying to improve it.

						As of early 2015, the opinion was that some work still needed to be done concerning the resources and support for _S. cerevisiae_ on BaseSpace in general .

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						#### info MACS for ChIP-seq analysis

						- it removes redundant reads
						- fragment size is a parameter you can set or let it determine
						- preferably calculates peak enrichment using local background normalization
						- can also estimate an empirical false discovery rate (FDR) for every peak if control available
						See the two publications about MACS listed second from the top in [the literature list](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/literature/#chip-seq).

					</script>
				</section>


				<section data-markdown>
					<script type="text/template">
						##Exploring results

						usually examine at peaks in viewers like IGV, IGB, SeqMonk

					</script>
				</section>



				<section data-markdown>
					<script type="text/template">
						##Other types of analyses with ChIP-seq data

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##Find Overlaps Of Chip-Seq Peaks In Multiple Samples

						use ChIPpeakAnno in R

						or

						intersectBed from Bedtools suite (popular command line tool)

						[tutorial on using to compare large sets of genomic features](http://ged.msu.edu/angus/tutorials-2011/bedtools-tutorial.html), a.k.a. interval operations


					</script>
				</section>




				<section data-markdown>
					<script type="text/template">
						##Extracting binding site information

					</script>
				</section>

				<section data-transition="slide" data-background="#4d7e65" data-background-transition="zoom">
					<img data-src="http://image.slidesharecdn.com/wednesdaysession10-timothybailey1-131030195105-phpapp01/95/motifbased-analysis-of-chipseq-data-timothy-bailey-24-638.jpg" alt="motifs">
					<br>
					motif discovery using MEME or other tools.

				</section>

				<section data-markdown>
					<script type="text/template">
						##Using CEAS to see what features are nearby

						[Day 7: ChIP-seq: Peak Predictions and Cis-regulatory Element Annotations](http://ged.msu.edu/angus/tutorials-2011/day7.html)

						Use CEAS to visualize the average ChIP enrichment signals over specific genomic features.



					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						####Determining the average ChIP-seq signal over promoters

						brings ups to another tool for your toolbelt...


					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						###IPython Notebooks

						[Link](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/in%20the%20wild%202/)

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						##bias in ChIP-seq
						- [Widespread misinterpretable ChIP-seq bias in yeast. Park et al PLoS One. 2013 PMID: 24349523](http://www.ncbi.nlm.nih.gov/pubmed/24173036#cm24173036_3919)

						- [Fan X and Struhl, 2009 "Where Does Mediator Bind In Vivo?" PLOS One. Published: April 03, 2009DOI: 10.1371/journal.pone.0005029](http://www.plosone.org/article/related/info%3Adoi%2F10.1371%2Fjournal.pone.0005029)

						- [Teytelman et al, 2013. Highly expressed loci are vulnerable to misleading ChIP localization of multiple unrelated proteins. Proc Natl Acad Sci U S A. 2013 Nov 12;110(46):18602-7. doi: 10.1073/pnas.1316064110.](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3831989/)

						See [literature list](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/literature/#bias-issues) for more information.
					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						####One approach to mitigation has been eliminating formaldehyde cross-linking
						[High-resolution mapping of transcription factor binding sites on native chromatin. Sivakanthan Kasinathan,Guillermo A Orsi, Gabriel E Zentner, Kami Ahmad & Steven Henikoff. Nature Methods 11, 203â€“209 (2014) doi:10.1038/nmeth.2766](http://www.nature.com/nmeth/journal/v11/n2/full/nmeth.2766.html)

						- called "ORGANIC" or native method

						- also obviates the need to normalize ChIP samples to input, which can be costly for large genomes due to the need for deep whole-genome sequence coverage

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						###Git, Github, and Gists for Version Control Software and collaboration

						[Link](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/in%20the%20wild%203/)

					</script>
				</section>


				<section data-markdown>
					<script type="text/template">
						###Regular Expressions

						[Link to hands-on session](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/in%20the%20wild%201/)

					</script>
				</section>

				<section data-markdown>
					<script type="text/template">
						###IPython Notebooks

						[Link](http://feng-lab-data-science-toolbox-and-chip-seq.readthedocs.org/en/latest/in%20the%20wild%202/)

					</script>
				</section>










				<section data-markdown>
					<script type="text/template">
						####Next session...
						Hands-on ChIP-seq


					</script>
				</section>
















				<section style="text-align: left;">
					<h1>THE END</h1>
					<p>

					</p>
				</section>







			</div>

		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>

			// Full list of configuration options available at:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				transition: 'slide', // none/fade/slide/convex/concave/zoom

				// Optional reveal.js plugins
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, condition: function() { return !!document.querySelector( 'pre code' ); }, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true },
					{ src: 'plugin/notes/notes.js', async: true }
				]
			});

		</script>

	</body>
</html>
